spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -492,9 +495,26 @@ public void doCommit(long epochId, WriterCommitMessage[] messages) {
 
   public static class TaskCommit implements WriterCommitMessage {
     private final DataFile[] taskFiles;
+    private long bytesWritten = 0L;
+    private long recordsWritten = 0L;
 
     TaskCommit(DataFile[] taskFiles) {
       this.taskFiles = taskFiles;
+      for (DataFile dataFile : taskFiles) {
+        this.bytesWritten += dataFile.fileSizeInBytes();
+        this.recordsWritten += dataFile.recordCount();
+      }
+    }
+
+    // Reports bytesWritten and recordsWritten to the Spark output metrics.
+    // Can only be called in executor.
+    void reportsToOutputMetrics() {
```

Comment:
I think `reportOutputMetrics` is a slightly better name.

Reply:
done